src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1018,6 +1021,21 @@ public static Double count (Object o) {
     }
 
     /**
+     * A node is considered non-empty if it is convertible into a string with a greater-than zero length.
+     *
+     * @param o NodeSet to evaluate. Throws if not a NodeSet
+     * @return the number of non-empty nodes in argument node-set.
+     */
+    public static Double countNonEmpty (Object o) {
+        if (o instanceof XPathNodeset) {
+            return (double) ((XPathNodeset) o).nonEmptySize();
+        } else {
```

Comment:
else after a return is not needed (nor any more clear, I would claim)

Reply:
Usually that's the pattern I prefer as well. Must have been late when I made this change 😄